request-manager/spec/common_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020, Square, Inc.
+
+package spec
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/go-test/deep"
+)
+
+const (
+	specsDir = "../test/specs/"
+)
+
+var (
+	printf = func(s string, args ...interface{}) { fmt.Printf(s, args...) }
+)
+
+func runSequenceCheckOn(check SequenceCheck, filename string) error {
+	filepath := specsDir + filename
+	allSpecs, err := ParseSpec(filepath, printf)
+	if err != nil {
+		return err
+	}
+	for _, sequence := range allSpecs.Sequences {
+		if err := check.CheckSequence(*sequence); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func runNodeCheckOn(check NodeCheck, filename string) error {
```

Comment:
Why don't you just manually create a NodeSpec struct for each test, instead of parsing in a file? That has the benefit of not relying on ParseSpec and of putting the example you're testing right next to the test, which makes it easier to see what exactly is being tested. It's also more "unit test"-like.

The one downside is that you won't be testing whether what you expect in the yaml actually corresponds to what you're writing in the NodeSpec struct. However in most cases the conversion from yaml to struct is obvious, and we can assume the yaml parser acts correctly. It's a good idea to also run some larger "integration" tests which go from yaml through parsing and checking (both static + graph). However I think that can happen later in a different place.